index.js

```diff
@@ -1,25 +1,54 @@
 'use strict';
-
+const Joi = require('joi');
 const Hoek = require('hoek');
+const Pkg = require('./package.json');
 let Pg = require('pg');
 
-
-const DEFAULTS = {
-    connectionString: undefined,
-    native: false,
-    attach: 'onPreHandler',
-    detach: 'tail'
+const HAPI_SERVER_EVENTS = [
+    'onRequest',
+    'onPreAuth',
+    'onPostAuth',
+    'onPreHandler',
+    'onPostHandler',
+    'onPreResponse',
+    'tail'
+];
+
+const internals = {
+    options: Joi.object({
+        connectionString: Joi.string().required(),
+        native: Joi.boolean().default(false),
+        attach: Joi.string().valid(HAPI_SERVER_EVENTS).default('onPreHandler'),
+        detach: Joi.string().valid(HAPI_SERVER_EVENTS).default('tail')
+    })
 };
 
-
 exports.register = function (server, options, next) {
 
-    const config = Hoek.applyToDefaults(DEFAULTS, options);
+    const validateOptions = internals.options.validate(options);
+
+    if (validateOptions.error) {
+        return next(validateOptions.error);
+    }
+
+    const config = Hoek.applyToDefaults(validateOptions.value, options);
 
     if (config.native) {
         Pg = require('pg').native;
     }
 
+    Pg.connect(config.connectionString, (err, client) => {
+
+        console.log(err);
+        console.log(client);
+        if (err) {
+            server.log([Pkg.name], `Connection Error ${JSON.stringify(err)}`);
+            return;
+        }
+        server.expose('pgClient', client);
```

Comment:
We're only exposing a client in this case and not the `done` function. The `done` function is what will release the client back to the pool. In this way, you've essentially _leaked_ a single dedicated client that won't ever get returned to the pool.